version: '3.9'  # Указываем версию
services:
  postgres:
    image: postgres:latest
    ports:
      - '5432:5432'
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: postgres
      POSTGRES_PORT: 5432

    volumes:
      - ./postgres-data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 5s
      timeout: 3s
      retries: 3

    networks:
      - pet_project

  django_app:
    image: vmonin/django_skypro_petproject:$GITHUB_REF_NAME-$GITHUB_RUN_ID
    ports:
      - "8000:8000"
    depends_on:
      postgres:
        condition: service_healthy
      migrations:
        condition: service_completed_successfully
    command: [ "python", "manage.py", "runserver", "0.0.0.0:8000"] #  "45.142.211.30:8000"

    networks:
      - pet_project

  migrations:
    image: vmonin/django_skypro_petproject:$GITHUB_REF_NAME-$GITHUB_RUN_ID
    depends_on:
      postgres:
        condition: service_healthy
    command: [ "python", "manage.py", "migrate" ]

    networks:
      - pet_project

networks:
  pet_project:
